lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -953,11 +941,14 @@ def _relation_hosts(self, relation) -> dict:
             A dictionary that maps unit names to unit addresses for
             the specified relation.
         """
-        return {
-            unit.name: relation.data[unit].get("prometheus_scrape_host")
-            for unit in relation.units
-            if relation.data[unit].get("prometheus_scrape_host")
-        }
+        hosts = {}
+        for unit in relation.units:
+            unit_name = relation.data[unit].get("prometheus_scrape_unit_name")
```

Comment:
This is not handling the case in which the other side is not adding the unit name, which is the case for every charm out there atm :-)

I feel we are being a touch too cavalier about backwards compatibility here.

Reply:
This change does break compatibility with previous version. Charms will need to updated to the current version of the charm library. If ùnit_name` is not set then that scrape target is not scrapped. You mean we should allow scrapping of charms that do not set unit names ? If so in this case should the unit names be taken from the event as was previously done ?

Reply:
I mean that, if the consumer is talking with an older version of the producer that sends relation data, it should not break. In such a case, keeping the old behavior as fallback is fine.

Reply:
Done.